src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -232,6 +240,48 @@ protected void deleteLocalApplicationResourceFile(AppDeploymentRequest appDeploy
 		}
 	}
 
+	protected Map<String, String> getEnvironmentVariables(String deploymentId, AppDeploymentRequest request) {
```

Comment:
and here

Reply:
The fact that the class itself has package access (e.g. no modifier) means that only subclasses from the same package can access its fields/methods. The protected won't extend or change the package access. (https://stackoverflow.com/questions/215497/what-is-the-difference-between-public-protected-package-private-and-private-in).

Anyway the PR looks good to me and can be merged.